src/lib/themes.js

```diff
@@ -89,15 +89,16 @@ export const themesEqual = (themeA, themeB) => {
     const alphaInA = "a" in colorA;
     const alphaInB = "a" in colorB;
     if (alphaInA !== alphaInB) {
-      return false;
-    }
-    if (
-      alphaInA &&
-      alphaInB &&
-      Math.abs(colorA.a - colorB.a) > alphaEqualityTolerance
-    ) {
-      return false;
-    }
+      if ((alphaInA && colorA.a !== 1) || (alphaInB && colorB.a !== 1)) {
```

Comment:
The logic of these conditions spread over multiple lines look complicated.

Since absence of `a` is now treated like `a: 1`, use the nullish coalescing operator to get the desired (numeric) value for `alphaA` and `alphaB`. Then you'll only need one if-condition like `if (Math.abs(alphaA - alphaB) > alphaEqualityTolerance) {`

Note that I'm suggesting the nullish coalescing operator instead of OR, to make sure that `0` stays `0` (i.e. transparent) and doesn't become `1` (opaque). Considering these edge cases, it would also be useful to check if there are existing tests for alpha being undefined, 0, 1, in between (e.g. 0.5), and if they don't exist yet, add them.